libr/util/buf.c

```diff
@@ -779,33 +779,24 @@ R_API bool r_buf_append_buf_slice(RBuffer *b, RBuffer *a, ut64 offset, ut64 size
 // length depends on the first '\0' found in the buffer.
 R_API char *r_buf_get_string(RBuffer *b, ut64 addr) {
 	const ut8 *needle = NULL;
-	ut8 tmp[16];
+	ut8 tmp[128] = {0};
 	ut64 sz = 0;
-	int r = r_buf_read_at (b, addr + sz, tmp, sizeof (tmp));
+	r_buf_read_at (b, addr, tmp, sizeof (tmp) - 1);
+	int r = strlen ((const char *)tmp);
 	while (r >= 0) {
 		needle = r_mem_mem (tmp, r, (ut8 *)"\x00", 1);
 		if (needle) {
 			sz += (needle - tmp);
 			break;
 		}
 		sz += r;
-		r = r_buf_read_at (b, addr + sz, tmp, sizeof (tmp));
+		r = r_buf_read_at (b, addr + sz, tmp + sz, sizeof (tmp) - sz);
```

Comment:
this seems wrong... the purpose of tmp was to read a possibly long string chunk-by-chunk, until you find the terminator char. At that point, once you know the length of the string, you can read exactly `sz` bytes from the buffer.

I think the change you made could result in overflows of `tmp`.